src/Kestrel.Core/Internal/IDefaultHttpsProvider.cs

```diff
@@ -1,10 +1,21 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Security.Cryptography.X509Certificates;
+
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
 {
     public interface IDefaultHttpsProvider
     {
+        /// <summary>
+        /// Returns the default certificate, if available, otherwise null.
+        /// </summary>
+        X509Certificate2 Certificate { get; }
```

Comment:
When does this need to be set?

Reply:
You found it over at https://github.com/aspnet/KestrelHttpServer/pull/2186/files#diff-590565eed2ca0d80acae41edb7fde1a7R230. Continuing the discussion there.